uaclient/cli.py

```diff
@@ -524,6 +533,13 @@ def _attach_with_token(
     except util.UrlError as exc:
         with util.disable_log_to_console():
             logging.exception(exc)
+
+        # Check if we have used a past machine-token file.
+        # If we had, delete this file, otherwise the system
+        # will think that it attached to the past subscription.
+        if os.path.exists(defaults.MACHINE_TOKEN_FILE):
+            util.del_file(defaults.MACHINE_TOKEN_FILE)
+
```

Comment:
This isn't checking the PAST_MACHINE_TOKEN
but as my suggestion above, we should use a UAConfig.data_paths key for this
```suggestion
           cfg.delete_cache("machine-token-saved")   # only deletes if cached json.save file exists
```

Reply:
Yes, that was incorrect, but we still need to delete the `machine-token` here. The scenario is that we failed to perform the attach but we have copied the `machine-token.json.save` into `machine-token.json`, which will cause uaclient to think that the system is attached, even though the attach has failed.